parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java

```diff
@@ -124,10 +125,20 @@ public Metrics metrics() {
     return ParquetUtil.footerMetrics(writer.getFooter(), model.metrics(), metricsConfig);
   }
 
+  /**
+   * Returns the number of bytes written by this writer.
+   * NOTE: call {@link ParquetWriter#close()})} prior this method to get the exact number of bytes.
+   *
+   * @return the number of bytes written by this writer.
```

Comment:
This isn't quite accurate. It is the number of bytes written only after the writer is closed. How about `@return length of the file produced by this writer, approximate until after closed`